internal/test/e2e/kubectl_test.go

```diff
@@ -1,3 +1,5 @@
+// +build e2e
```

Comment:
What's this for?

Reply:
It prevents the e2e tests being run at the same time as the kubebuilder / unit tests.
Previously, the e2e tests had been skipped....almost by accident....because `make test` doesn't supply either `--context` or `--kind`.
This build tag ensures that e2e tests are only run if we supply a `go test --tag` argument

Reply:
Replacing this with a simple `--e2e-enabled` flag.